server/event/dispatch/operation.go

```diff
@@ -129,6 +137,45 @@ func (o *Operation) dispatch(wfeb wfv1.WorkflowEventBinding, nameSuffix string)
 	return nil, nil
 }
 
+func (o *Operation) populateWorkflowMetadata(wf *wfv1.Workflow, metadata *metav1.ObjectMeta) error {
+	if len(metadata.Name) > 0 {
+		evalName, err := o.evaluateStringExpression(metadata.Name, "name")
+		if err != nil {
+			return err
+		}
+		wf.SetName(evalName)
+	}
+	for labelKey, labelValue := range metadata.Labels {
+		evalLabel, err := o.evaluateStringExpression(labelValue, fmt.Sprintf("label \"%s\"", labelKey))
+		if err != nil {
+			return err
+		}
+		wf.Labels[labelKey] = evalLabel
```

Comment:
I think you might need to check if `wf.Labels == nil` here

Reply:
Fixed. Did the same for annotations.